vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -44,6 +56,16 @@ def _filter_isin(series: pd.Series, value: MultiValueType) -> pd.Series:
     return series.isin(value)
 
 
+def _filter_between_date(series: pd.Series, value: List[str]) -> pd.Series:
+    date_series, date_value = [_convert_to_date(value=elem) for elem in (series, value)]
```

Comment:
Also just an FYI because I think we should get rid of this, but this line is super confusing 😅 When it's just two unrelated elements like this I would never do it with a loop because trying to unpack this assignment in my head took way longer than if it was just written like this:
```suggestion
    date_series, date_value = _convert_to_date(series), _convert_to_date(value)
```